Ports/quake3/patches/0001-Meta-Refactor-Makefile-to-support-Serenity.patch

```diff
@@ -0,0 +1,133 @@
+From 702293dcfc074c7d0bec034bcc8d477ff1799cf4 Mon Sep 17 00:00:00 2001
+From: Jesse Buhagiar <jooster669@gmail.com>
+Date: Fri, 25 Mar 2022 09:39:21 +1100
+Subject: [PATCH 1/8] Meta: Refactor Makefile to support Serenity
+
+---
+ Makefile | 37 +++++++++++++++++++------------------
+ 1 file changed, 19 insertions(+), 18 deletions(-)
+
+diff --git a/Makefile b/Makefile
+index 9e535559..027bd264 100644
+--- a/Makefile
++++ b/Makefile
+@@ -3,8 +3,8 @@
+ #
+ # GNU Make required
+ #
+-COMPILE_PLATFORM=$(shell uname | sed -e 's/_.*//' | tr '[:upper:]' '[:lower:]' | sed -e 's/\//_/g')
+-COMPILE_ARCH=$(shell uname -m | sed -e 's/i.86/x86/' | sed -e 's/^arm.*/arm/')
++COMPILE_PLATFORM=serenity
++COMPILE_ARCH=amd64
+ 
+ #arm64 hack!
+ ifeq ($(shell uname -m), arm64)
+@@ -35,13 +35,13 @@ ifndef BUILD_GAME_QVM
+   BUILD_GAME_QVM   =
+ endif
+ ifndef BUILD_BASEGAME
+-  BUILD_BASEGAME =
++  BUILD_BASEGAME =1
+ endif
+ ifndef BUILD_MISSIONPACK
+-  BUILD_MISSIONPACK=
++  BUILD_MISSIONPACK=0
+ endif
+ ifndef BUILD_RENDERER_OPENGL2
+-  BUILD_RENDERER_OPENGL2=
++  BUILD_RENDERER_OPENGL2=0
+ endif
+ ifndef BUILD_AUTOUPDATER  # DON'T build unless you mean to!
+   BUILD_AUTOUPDATER=0
+@@ -164,39 +164,39 @@ GENERATE_DEPENDENCIES=1
+ endif
+ 
+ ifndef USE_OPENAL
+-USE_OPENAL=1
++USE_OPENAL=0
+ endif
+ 
+ ifndef USE_OPENAL_DLOPEN
+-USE_OPENAL_DLOPEN=1
++USE_OPENAL_DLOPEN=0
+ endif
+ 
+ ifndef USE_CURL
+-USE_CURL=1
++USE_CURL=0
+ endif
+ 
+ ifndef USE_CURL_DLOPEN
+   ifdef MINGW
+     USE_CURL_DLOPEN=0
+   else
+-    USE_CURL_DLOPEN=1
++    USE_CURL_DLOPEN=0
+   endif
+ endif
+ 
+ ifndef USE_CODEC_VORBIS
+-USE_CODEC_VORBIS=1
++USE_CODEC_VORBIS=0
+ endif
+ 
+ ifndef USE_CODEC_OPUS
+-USE_CODEC_OPUS=1
++USE_CODEC_OPUS=0
+ endif
+ 
+ ifndef USE_MUMBLE
+-USE_MUMBLE=1
++USE_MUMBLE=0
+ endif
+ 
+ ifndef USE_VOIP
+-USE_VOIP=1
++USE_VOIP=0
+ endif
+ 
+ ifndef USE_FREETYPE
+@@ -506,7 +506,7 @@ ifeq ($(PLATFORM),darwin)
+     # If CC is already set to something generic, we probably want to use
+     # something more specific
+     ifneq ($(findstring $(strip $(CC)),cc gcc),)
+-      CC=
++      CC=x86_64-pc-serenity-gcc
+     endif
+ 
+     ifndef CC
+@@ -817,11 +817,11 @@ ifeq ($(PLATFORM),freebsd)
+ else # ifeq freebsd
+ 
+ #############################################################################
+-# SETUP AND BUILD -- OPENBSD
++# SETUP AND BUILD -- SERENITY
+ #############################################################################
+ 
+-ifeq ($(PLATFORM),openbsd)
+-
++ifeq ($(PLATFORM),serenity)
++  CC=x86_64-pc-serenity-gcc
```

Comment:
same for these triples. do we have anything in `Ports/.port_include.sh` or `Ports/.hosted_defs.sh` that make up a $SERENITY_TRIPLE or some such? This also means the port won't work at all for clang, the way gcc is hard coded.

Reply:
Looks like I can use the following: `CC="${SERENITY_ARCH}-pc-serenity-${tool}"`! Neato!

Reply:
Right, unless you want to use the Clang Toolchain 😬. $CC should be set by .port_include.sh and .hosted_defs.sh already, can we just use that? 😅

Reply:
Does `clang` have a different tuple or something? I've never actually used the toolchain before

Reply:
@ADKaster Okay, I've removed all the `CC` instances to let the ports include script do it's thing instead.